### ISSUE TITLE
Cannot use positional argument after named argument during unpacking

### ISSUE THREAD

**Issue by Kehet** [spatie/data-transfer-object#298]
I found following edge case:

```php
class IssueDTO extends DataTransferObject
{
    public ?ReactionRollupDTO $reactions;
}
```

```php
class ReactionRollupDTO extends DataTransferObject
{
    public int $total_count;

    #[MapFrom('+1')]
    #[MapTo('+1')]
    public int $plus_one;

    #[MapFrom('-1')]
    #[MapTo('-1')]
    public int $minus_one;

    public int $laugh;
}
```

```php
$reactions = [
    'total_count' => 0,
    '+1' => 0,
    -1 => 0,
    'laugh' => 0,
];

// works as expected
dump(new ReactionRollupDTO($reactions));

$issue = [
    'reactions' => $reactions,
];

// Error: Cannot use positional argument after named argument during unpacking
dump(new IssueDTO($issue));
```

Example is how `json_decode` decodes json by default. Not exactly ideal but it is what it is..

I'm aware this could propably be handled with custom caster but my question is if this is something that should be handled in package?

**Comment by aidan-casey** [comment]
@Kehet - I'm not entirely sure I'm understanding this issue correctly. Would you be able to create a PR with a failing test so we can begin to dig into this?

**Comment by Kehet** [comment]
@aidan-casey I refined this problem bit and wrote two tests to catch it https://github.com/spatie/data-transfer-object/pull/302

**Comment by aidan-casey** [comment]
Thank you, @Kehet. I will look into this.

**Comment by Kehet** [comment]
@aidan-casey This works in my codebase, thank you

**Comment by aidan-casey** [comment]
Fixed in `3.9.1`.
